frontend/src/components/worksheets/items/ImageItem.js

```diff
@@ -1,23 +1,49 @@
 import * as React from 'react';
-import Immutable from 'seamless-immutable';
-import { worksheetItemPropsChanged } from '../../../util/worksheet_utils';
+import { withStyles } from '@material-ui/core/styles';
+import * as Mousetrap from '../../../util/ws_mousetrap_fork';
+import BundleDetail from '../BundleDetail';
 
 class ImageItem extends React.Component {
     /** Constructor. */
     constructor(props) {
         super(props);
-        this.state = Immutable({});
+        this.state = {
+            showDetail: false,
+            bundleInfoUpdates: {},
+        };
     }
 
+    receiveBundleInfoUpdates = (update) => {
+        let { bundleInfoUpdates } = this.state;
+        // Use object spread to update.
+        bundleInfoUpdates = { ...bundleInfoUpdates, ...update };
+        this.setState({ bundleInfoUpdates });
```

Comment:
I think this would be clearer
```suggestion
        this.setState({ bundleInfoUpdates: { ...bundleInfoUpdates, ...update } });
```